### DISCUSSION TITLE
What's the most computation intensive part of create program?

### DISCUSSION THREAD

**Discussion by lin-ycv** [visose/Robots#297] [Q&A] [answered]
i'm curious as to what makes a few hundred targets take 5 seconds to calculate
![image](https://github.com/user-attachments/assets/249e9552-2f71-4c93-a8c7-504a9b748588)

I've done programs that have a lot more targets, and it only takes 2 or so seconds.

Does a detailed robot/tool mesh model causes increase in time? or does long distance movements increase time?
What tips are there to minmize the computational time.

**Comment by visose** [comment, marked as answer]
Linear motions will take longer than Joint motions. Longer distance traveled in linear motions will also take longer, irrespective of the number of targets. Some robots use a numerical IK solver, which takes longer than the analytical ones, but this is not the case for KUKA which I think you are using.
You can reduce the time it takes to calculate by increasing the Step size input (S). You can keep it a higher value while making changes and lower it back to 1 mm at the very end before exporting the program.
